Sources/AblyChat/Message.swift

```diff
@@ -171,9 +171,13 @@ extension Message: JSONObjectDecodable {
                 values: summaryJson
             )
         }
+        let rawAction = try jsonObject.stringValueForKey("action")
+        guard let action = MessageAction(rawValue: rawAction) else {
+            throw JSONValueDecodingError.failedToDecodeFromRawValue(rawAction).toInternalError()
```

Comment:
Also would be good to have a convenience method à la `rawRepresentableValueForKey`

Reply:
@lawrence-forooghian let me know if this is important and/or create issue from your comment, I'm now resolving this.